source/core/parameters.cc

```diff
@@ -499,6 +499,99 @@ namespace Parameters
       prm.get_double("cahn hilliard mobility constant");
   }
 
+  void
+  ConstrainSolidDomain::declare_parameters(
+    dealii::ParameterHandler &prm,
+    const unsigned int        number_of_constraints)
+  {
+    prm.enter_subsection("constrain solid domain");
+    {
+      prm.declare_entry(
+        "enable",
+        "false",
+        Patterns::Bool(),
+        "Enable/disable (true/false) the solid domain constraining feature.");
+      prm.declare_entry(
+        "number of constraints",
+        Utilities::int_to_string(number_of_constraints),
+        Patterns::Integer(),
+        "Number of solid constraints (maximum of 1 per fluid).");
+      // Resize vectors
+      this->fluid_ids.resize(number_of_constraints);
+      this->temperature_min_values.resize(number_of_constraints);
+      this->temperature_max_values.resize(number_of_constraints);
+
+      // Declare default entries
+      for (unsigned int c_id = 0; c_id < number_of_constraints; c_id++)
+        {
+          prm.enter_subsection("constraint " + std::to_string(c_id));
+          {
+            declare_default_entries(prm);
+          }
+          prm.leave_subsection();
+        }
+    }
+
+    prm.leave_subsection();
+  }
+
+  void
+  ConstrainSolidDomain::declare_default_entries(dealii::ParameterHandler &prm)
+  {
+    prm.declare_entry("fluid id",
+                      "0",
+                      Patterns::Integer(),
+                      "Identifier of the fluid material that is constrained.");
+    prm.declare_entry("min temperature",
+                      "-999",
+                      Patterns::Double(),
+                      "Minimum temperature value of the fluid for it be "
+                      "considered as a solid.");
+    prm.declare_entry("max temperature",
+                      "0",
+                      Patterns::Double(),
+                      "Maximum temperature value of the fluid for it be "
+                      "considered as a solid.");
+  }
+
+  void
+  ConstrainSolidDomain::parse_parameters(dealii::ParameterHandler &prm)
+  {
+    prm.enter_subsection("constrain solid domain");
+    {
+      this->enable                = prm.get_bool("enable");
+      this->number_of_constraints = prm.get_integer("number of constraints");
+
+      // Resize vectors
+      this->fluid_ids.resize(number_of_constraints);
+      this->temperature_min_values.resize(number_of_constraints);
+      this->temperature_max_values.resize(number_of_constraints);
+
+      // Parse parameters for each constraint
+      for (unsigned int c_id = 0; c_id < number_of_constraints; c_id++)
+        {
+          prm.enter_subsection("constraint " + std::to_string(c_id));
+          {
+            parse_constraint_parameters(prm, c_id);
+          }
+          prm.leave_subsection();
+        }
+      prm.leave_subsection();
+    }
```

Comment:
Be careful here, you will have declared a maximum of 1 constraints, so you should throw an error if number_of_contraints is larger than 1.